### PR TITLE
Chore/dataguid org test

### PR DESCRIPTION
**Pull request by jingh8**
Jira Ticket: [PXP-7072](https://ctds-planx.atlassian.net/browse/PXP-7072)

Test https://dataguids.org
[Test plan for dataguids.org](https://github.com/uc-cdis/gen3-qa/blob/master/docs/testplans/environment-specific/dataguidsorg-test-plan.md)

### CODE REVIEW

**Review by haraprasadj — APPROVED** [review]
+1

File: suites/portal/dataguidOrgTest.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+correctGuids.add(['0000047f-772f-4241-980d-1f667686fe60']); //adding guids for "NCI CRDC" which doesn't support prefix
+correctGuids.add(['dg.ANV0/0000b4b4-2af4-42e2-9bfa-6fd11e5fb97a']); //adding guids for "AnVIL"
+
+let nonexistentGuids = new DataTable(['nguids']);
+nonexistentGuids.add(['dg.ABCD/0000b4b4-2af4-42e2-9bfa-6fd11e5fb97a']); //adding guids with wrong prefix
+nonexistentGuids.add(['0000b456-3r56-1dr3-0rt4-6fd11e5fb97a']); //adding non-existent guids
+
+//Pass dataTable to Data()
+//Test resolving guids
+Data(correctGuids).Scenario('resolveguids', (I, current) => {
```
Comment (themarcelor) [inline suites/portal/dataguidOrgTest.js:17]: Minor change.
Can you please annotate the scenarios with something like @dataguids?

Comment (jingh8) [reply suites/portal/dataguidOrgTest.js:17]: Sure

**Review by atharvar28 — APPROVED** [review]

File: suites/portal/dataguidOrgTest.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+correctGuids.add(['dg.ANV0/0000b4b4-2af4-42e2-9bfa-6fd11e5fb97a']); // adding guids for "AnVIL"
+
+const nonexistentGuids = new DataTable(['nguids']);
+nonexistentGuids.add(['dg.ABCD/0000b4b4-2af4-42e2-9bfa-6fd11e5fb97a']); // adding guids with wrong prefix
+nonexistentGuids.add(['0000b456-3r56-1dr3-0rt4-6fd11e5fb97a']); // adding non-existent guids
+
+// Pass dataTable to Data()
+// Test resolving guids
+Data(correctGuids).Scenario('Resolve guids with different prefixes or without prefix @dataguids', (I, current) => {
+  I.amOnPage('https://dataguids.org');
```
Comment (themarcelor) [inline suites/portal/dataguidOrgTest.js:18]: This should target a mutated jenkins environment in the CI Pipeline (it should not have the hardcoded address of the actual PROD environment).

File: suites/portal/dataguidOrgTest.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  I.fillField('#guidval', current.nguids);
+  I.click('#resolveit');
+  I.waitForText(current.nguids, 2, '#resolverresult');
+  I.see(`Data GUID "${current.nguids}" not found.`);
+});
+
+// Test DRS endpoint
+Data(correctGuids).Scenario('Test if DRSendpoint resolve the guids correctly @dataguids', (I, current) => {
+  I.amOnPage(`https://dataguids.org/index/ga4gh/drs/v1/objects/${current.guids}`);
```
Comment (themarcelor) [inline suites/portal/dataguidOrgTest.js:36]: This should target a mutated jenkins environment in the CI Pipeline (it should not have the hardcoded address of the actual PROD environment).

You can just do this and CodeceptJS will take care of injecting the jenkins environment's hostname in there:
```
I.amOnPage(`/index/ga4gh/drs/v1/objects/${current.guids}`);
```

File: suites/portal/dataguidOrgTest.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  I.see(`"id":"${current.guids}"`);
+});
+
+// Nagative DRS endpoint test
+Data(nonexistentGuids).Scenario('Negativetest DRSendpoint with non-existent guids @dataguids', (I, current) => {
```
Comment (themarcelor) [inline suites/portal/dataguidOrgTest.js:41]: @jingh8 , please adopt the new CodeceptJS 3 syntax.
It's super easy, you just need to replace this:
```
 (I, current)
```
with this:
```
 ({ I, current })
```

File: suites/portal/dataguidOrgTest.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+Data(correctGuids).Scenario('Resolve guids with different prefixes or without prefix @dataguids', (I, current) => {
+  I.amOnPage('https://dataguids.org');
+  I.fillField('#guidval', current.guids);
+  I.click('#resolveit');
+  I.waitForText(current.guids, 2, '#resolverresult');
+  I.see(`"id": "${current.guids}"`);
+});
+
+// Nagative resolving guids test
+Data(nonexistentGuids).Scenario('Negativetest resolving for non-exitent guids @dataguids', (I, current) => {
```
Comment (themarcelor) [inline suites/portal/dataguidOrgTest.js:26]: @jingh8 , please adopt the new CodeceptJS 3 syntax.
It's super easy, you just need to replace this:
```
 (I, current)
```
with this:
```
 ({ I, current })
```

File: suites/portal/dataguidOrgTest.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+correctGuids.add(['0000047f-772f-4241-980d-1f667686fe60']); // adding guids for "NCI CRDC" which doesn't support prefix
+correctGuids.add(['dg.ANV0/0000b4b4-2af4-42e2-9bfa-6fd11e5fb97a']); // adding guids for "AnVIL"
+
+const nonexistentGuids = new DataTable(['nguids']);
+nonexistentGuids.add(['dg.ABCD/0000b4b4-2af4-42e2-9bfa-6fd11e5fb97a']); // adding guids with wrong prefix
+nonexistentGuids.add(['0000b456-3r56-1dr3-0rt4-6fd11e5fb97a']); // adding non-existent guids
+
+// Pass dataTable to Data()
+// Test resolving guids
+Data(correctGuids).Scenario('Resolve guids with different prefixes or without prefix @dataguids', (I, current) => {
```
Comment (themarcelor) [inline suites/portal/dataguidOrgTest.js:17]: @jingh8 , please adopt the new CodeceptJS 3 syntax.
It's super easy, you just need to replace this:
```
 (I, current)
```
with this:
```
 ({ I, current })
```

File: suites/portal/dataguidOrgTest.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  I.amOnPage('https://dataguids.org');
+  I.fillField('#guidval', current.guids);
+  I.click('#resolveit');
+  I.waitForText(current.guids, 2, '#resolverresult');
+  I.see(`"id": "${current.guids}"`);
+});
+
+// Nagative resolving guids test
+Data(nonexistentGuids).Scenario('Negativetest resolving for non-exitent guids @dataguids', (I, current) => {
+  I.amOnPage('https://dataguids.org');
```
Comment (themarcelor) [inline suites/portal/dataguidOrgTest.js:27]: This should target a mutated jenkins environment in the CI Pipeline (it should not have the hardcoded address of the actual PROD environment).

**Pull request merged**